src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -149,6 +158,46 @@ private String getOktaSessionToken() throws IOException {
         }
     }
 
+    private void multiProfileCreateIni() throws  IOException {
+
+        boolean result = false;
+        File okta = new File(System.getProperty("user.home") + "/.okta");
+        File profileIni = new File(System.getProperty("user.home") + "/.okta/.okta-profile-expiry");
+        if (!okta.exists()) {
+            System.out.println("creating Directory and File: " + okta.getName() + profileIni.getName());
+            try {
+                result = okta.mkdir() && profileIni.createNewFile();
+            } catch (SecurityException se) {
+                System.out.println("Permission too low");
+            }
+            if (result)
+                System.out.println("Okta profile expiry created");
+            else
+                System.out.println("Okta profile expiry was not created");
+        }
+    }
+    private boolean multiProfileIniExists() {
+        File okta = new File(System.getProperty("user.home") + "/.okta");
+        return(okta.exists());
+    }
+
+
+    private boolean profileDoesNotExistsOrExpired() throws IOException {
+        Instant start = Instant.now();
+        String profileStore = System.getProperty("user.home") + "/.okta/.okta-profile-expiry";
+        Ini ini = new Ini(new File(profileStore));
+        Set<String> activeSessions = ini.keySet();
+        if (oktaProfile == null) return true;
+        if (!activeSessions.contains(oktaProfile)) return true;
+        if (activeSessions.contains(oktaProfile)) {
+            Ini.Section profilesection = ini.get(oktaProfile);
+            String profileExpiry = profilesection.get("profile_expiry");
```

Comment:
This belongs in the **MultipleProfile** class. It's a detail specific to that. Ideally that class would hide the use of **Ini** from this class.

For example:

```java
MultipleProfile multipleProfile = new MultipleProfile(reader);
Optional<Profile> profile = multipleProfile.getProfile(oktaProfile);
if (profile.isPresent()) {
    Instant expiry = multipleProfile.getExpiry(oktaProfile);
    String roleArn = multipleProfile.getRoleArn(oktaProfile);
}
```